lsp-mode.el

```diff
@@ -2036,12 +2036,14 @@ CALLBACK - callback for the lenses."
     (add-hook 'lsp-on-idle-hook #'lsp--lens-idle-function nil t)
     (add-hook 'lsp-on-change-hook (lambda () (lsp--lens-schedule-refresh t)) nil t)
     (add-hook 'after-save-hook (lambda () (lsp--lens-schedule-refresh t)) nil t)
+    (add-hook 'before-revert-hook (lambda () (lsp-lens-hide)) nil t)
```

Comment:
nit: there is no need to wrap it in lambda, you could write directly `#'lsp-lens-hide`